osbuild-composer.spec

```
@@ -237,6 +249,28 @@ systemctl stop "osbuild-worker@*.service" "osbuild-remote-worker@*.service"
 # restart all the worker services
 %systemd_postun_with_restart "osbuild-worker@*.service" "osbuild-remote-worker@*.service"
 
+%if 0%{?rhel}
+%package osbuild-composer-cloud
+Summary:    The osbuild-composer cloud api
+Requires:   systemd
+Requires:   osbuild
+
+%description osbuild-composer-cloud
+The cloud api for osbuild-composer
+
+%files osbuild-composer-cloud
+%{_libexecdir}/osbuild-composer/osbuild-composer-cloud
+%{_unitdir}/osbuild-composer-cloud.socket
+%{_unitdir}/osbuild-composer-cloud.service
+
+%post osbuild-composer-cloud
+%systemd_post osbuild-composer-cloud.socket osbuild-composer-cloud.service
+
+%preun osbuild-composer-cloud
+systemctl --no-reload disable osbuild-composer-cloud.socket osbuild-composer-cloud.service
+systemctl stop "osbuild-composer-cloud.socket osbuild-composer-cloud.service"
```

```suggestion
%systemd_preun osbuild-composer-cloud.socket osbuild-composer-cloud.service
```

should be enough. The lines you used are just workaround for template services that are not used here.